### ISSUE TITLE
Help text is still displayed even if the field is hidden

### ISSUE THREAD

**Issue by Guydn** [twomice/com.joineryhq.profcond#36]
Is there a way to hide the help text (before ou after) for a field that is hidden ?

**Comment by twomice** [comment]
Yes, but it has to be explicitly shown/hidden with the `display` [state-property]

If you examine the HTML for the civicrm form, you'll see that pre- and post-help each appear in their own `<div>` element before and after the field. For example, if the profile contains pre- and post-help for custom field id=2, you'll see this in the form HTML:

```
<div class="crm-section helprow-custom_2-section helprow-pre" id="helprow-custom_2">
  <!-- Custom field "pre" help -->
</div>
<div class="crm-section editrow_custom_2-section form-item" id="editrow-custom_2">
  <!-- The custom field itself -->
</div>
<div class="crm-section helprow-custom_2-section helprow-post" id="helprow-custom_2">
  <!-- Custom field "post" help -->
</div>
```

You could hide the pre-help like so:
```php
// ...

              // [state-selector]:
              'div.helprow-custom_2-section.helprow-pre' => array(
                // [state-property]:
                'display' => 'hide',
              ),
// ...
```

Or you could hide _both_ pre- and post-help like so:
You could hide the pre-help like so:
```php
// ...

              // [state-selector]:
              'div.helprow-custom_2-section' => array( // Notice this selector doesn't specify .helprow-pre or .helprow-post
                // [state-property]:
                'display' => 'hide',
              ),
// ...
```

**Comment by twomice** [comment]
I'll close this for now, but feel free to re-open if you need more help!

**Comment by Guydn** [comment]
Thanks a lot (again :-)) !!

**Comment by wyde22** [comment]
Hi, my HTML is different of your description :)

here my HTML

```
<div class="crm-section activites_chorales crm-price-field-id-4">
   <div class="label">Activités chorales complètes</div>
   <div class="content content activit_chorales_compl_tes-content">
        <div class="price-set-row activit_choralies_compl_tes-row1">
              <input class="crm-form-radio" />
        </div>
   </div>
</div>

<span class="content description">Saisissez le nombre de jours</span>
```
my help text is on span with `content description` class. How can hide this element near of my custom field

**Comment by twomice** [comment]
Hi @wyde22 . The short answer is this: If you can come up with a jQuery selector that references the element, you can use that selector to show/hide it.

With only the HTML you've given, the only selector I can see that would work is 'div.content.description'. If that's not specific enough (i.e. if you have more than one such `<div>` on your page), you'll need to examine the html a little more to find a selector that is specific only to this one `<div>`.

BTW, if you're familiar with the Dev Tools javascript console in your browser, you might want to test your jQuery selectors easily right on the page, using javascript code like this, in the console:
```
CRM.$('div.content.description');
```
That will return all of the elements on the page that match that selector, so you can (more) quickly determine a selector that works for your needs.

Either way, you'll have to find a selector that identifies the relevant page elements, then you can show/hide using that selector.

Hope this helps!

**Comment by wyde22** [comment]
I'm finding a solution. The element HTML of help text is near of price field. i'm using this selector for hide or show the help text element : `\#priceset .crm-price-field-id-24+.description`. With this solution I can target the element precisely

it's works for me

**Comment by twomice** [comment]
Glad to hear it -- thanks @wyde22 !
